### DISCUSSION TITLE
encrypt backups before upload

### DISCUSSION THREAD

**Discussion by tcurdt** [coollabsio/coolify#1859] [Feature Requests]
It would be nice if the backups could be encrypted with `age` before they get uploaded to S3.

https://github.com/FiloSottile/age

**Comment by johnmaguire** [comment]
Encrypted backups are a must, IMO, especially considering that there are many secrets configured into Coolify (e.g. SMTP, S3 storage, Github apps, SSH tokens, etc.)

I don't feel particular on the encryption method used (e.g. GPG would be fine too), but age is great software.

**Comment by Seym0n** [comment]
Picking up this topic from the box; while `age` sounds great, we may need to spawn a separate process for it.

Since PHP provides out of the box the extension `ext-sodium`, which can be also used to encrypt backups, we may not need `age`.

If someone else has more ideas to that, feel free to speak it out,

**Reply by tcurdt** [reply]
The point of using `age` would be to allow decryption outside of the realm of `coolify`. Which feels like a nice secondary goal for the implementation. And getting encryption right is not always as easy as it sounds.
But I have moved on. So just my 2 cents.
